### PR TITLE
Protobuf3 grammar: support optional field

### PR DESCRIPTION
**Pull request by DmitryMaletin**
Add support for optional field as documented here: https://developers.google.com/protocol-buffers/docs/proto3#specifying_field_rules

### CODE REVIEW

File: protobuf3/Protobuf3.g4 [hunk 1]
```diff
@@ -55,9 +55,12 @@ optionName
   ;
 
 // Normal Field
+fieldLabel
+  : OPTIONAL | REPEATED
+  ;
 
 field
-  : ( REPEATED )? type_ fieldName EQ fieldNumber ( LB fieldOptions RB )? SEMI
+  : ( fieldLabel )? type_ fieldName EQ fieldNumber ( LB fieldOptions RB )? SEMI
```
Comment (KvanTTT) [inline protobuf3/Protobuf3.g4:63]: Please remove excess parentheses:

```suggestion
  : fieldLabel? type_ fieldName EQ fieldNumber ( LB fieldOptions RB )? SEMI
```

**Comment by KvanTTT** [comment]
Could you please add an example to `examples` subdirectory that covers suggested changes?

**Comment by DmitryMaletin** [comment]
> Could you please add an example to `examples` subdirectory that covers suggested changes?

Would it be enough  or is it better to modify/add  to examples/conformance.proto? 
Looks like singular fields there named as optional

**Comment by KvanTTT** [comment]
I think it's ok, thanks.

**Pull request merged**